src/celengine/galaxy.cpp

```diff
@@ -8,114 +8,357 @@
+constexpr float RADIUS_CORRECTION     = 0.025f;
+constexpr float MAX_SPIRAL_THICKNESS  = 0.06f;
+
+bool formsInitialized = false;
+
+constexpr std::size_t GalacticFormsReserve = 32;
+
+std::vector<std::optional<GalacticForm>> galacticForms;
+std::map<fs::path, std::size_t> customForms;
+
+Texture* galaxyTex = nullptr;
+Texture* colorTex  = nullptr;
+
 struct GalaxyTypeName
 {
     const char* name;
-    Galaxy::GalaxyType type;
+    GalaxyType type;
 };
 
-static GalaxyTypeName GalaxyTypeNames[] =
-{
-    { "S0",  Galaxy::S0 },
-    { "Sa",  Galaxy::Sa },
-    { "Sb",  Galaxy::Sb },
-    { "Sc",  Galaxy::Sc },
-    { "SBa", Galaxy::SBa },
-    { "SBb", Galaxy::SBb },
-    { "SBc", Galaxy::SBc },
-    { "E0",  Galaxy::E0 },
-    { "E1",  Galaxy::E1 },
-    { "E2",  Galaxy::E2 },
-    { "E3",  Galaxy::E3 },
-    { "E4",  Galaxy::E4 },
-    { "E5",  Galaxy::E5 },
-    { "E6",  Galaxy::E6 },
-    { "E7",  Galaxy::E7 },
-    { "Irr", Galaxy::Irr },
+constexpr const GalaxyTypeName GalaxyTypeNames[] =
+{
+    { "S0",  GalaxyType::S0 },
+    { "Sa",  GalaxyType::Sa },
+    { "Sb",  GalaxyType::Sb },
+    { "Sc",  GalaxyType::Sc },
+    { "SBa", GalaxyType::SBa },
+    { "SBb", GalaxyType::SBb },
+    { "SBc", GalaxyType::SBc },
+    { "E0",  GalaxyType::E0 },
+    { "E1",  GalaxyType::E1 },
+    { "E2",  GalaxyType::E2 },
+    { "E3",  GalaxyType::E3 },
+    { "E4",  GalaxyType::E4 },
+    { "E5",  GalaxyType::E5 },
+    { "E6",  GalaxyType::E6 },
+    { "E7",  GalaxyType::E7 },
+    { "Irr", GalaxyType::Irr },
 };
 
-
-static void GalaxyTextureEval(float u, float v, float /*w*/, unsigned char *pixel)
+void galaxyTextureEval(float u, float v, float /*w*/, unsigned char *pixel)
 {
-    float r = 0.9f - (float) sqrt(u * u + v * v );
+    float r = 0.9f - std::sqrt(u * u + v * v );
     if (r < 0)
         r = 0;
 
-    auto pixVal = (int) (r * 255.99f);
+    auto pixVal = static_cast<unsigned char>(r * 255.99f);
     pixel[0] = 255;//65;
     pixel[1] = 255;//64;
     pixel[2] = 255;//65;
     pixel[3] = pixVal;
 }
 
-static void ColorTextureEval(float u, float /*v*/, float /*w*/, unsigned char *pixel)
+
+void colorTextureEval(float u, float /*v*/, float /*w*/, unsigned char *pixel)
 {
-    unsigned int i = (u*0.5f + 0.5f)*255.99f; // [-1, 1] -> [0, 255]
+    unsigned int i = static_cast<unsigned int>((static_cast<float>(u)*0.5f + 0.5f)*255.99f); // [-1, 1] -> [0, 255]
 
     // generic Hue profile as deduced from true-color imaging for spirals
     // Hue in degrees
-    float hue = 25 * tanh(0.0615f * (27 - i));
-    if (i >= 28) hue += 220;
+    float hue = 25.0f * std::tanh(0.0615f * (27.0f - static_cast<float>(i)));
+    if (i >= 28) hue += 220.0f;
     //convert Hue to RGB
     float r, g, b;
     DeepSkyObject::hsv2rgb(&r, &g, &b, hue, 0.20f, 1.0f);
-    pixel[0] = (unsigned char) (r * 255.99f);
-    pixel[1] = (unsigned char) (g * 255.99f);
-    pixel[2] = (unsigned char) (b * 255.99f);
+    pixel[0] = static_cast<unsigned char>(r * 255.99f);
```

Comment:
i believe here we should use `uint8_t` instead of `unsigned char`, it will be more semantically correct